llama.cpp

```diff
@@ -7267,16 +7484,15 @@ static int llama_decode_internal(
 
     ggml_cgraph * gf = llama_build_graph(lctx, batch);
 
-    // the output is always the last tensor in the graph
-    struct ggml_tensor * res = gf->nodes[gf->n_nodes - 1];
-    GGML_ASSERT(strcmp(res->name, "result_output") == 0);
+    // get logits and embeddings
+    struct ggml_tensor * res = ggml_graph_get_tensor(gf, "result_output");
+    struct ggml_tensor * embeddings = ggml_graph_get_tensor(gf, "result_norm");
```

Comment:
Using `ggml_graph_get_tensor` is not recommended here because it will do a `strcmp` with the entire graph which can become noticeable in terms of speed. For now, we should be "poking" at the last few tensors to find what we need - not great, but will improve in the future